getpan/getpan/dependency.go

```diff
@@ -323,24 +216,91 @@ func (d *DependencyList) Resolve() error {
 	return nil
 }
 
-func (v *Dependency) Resolve() error {
-	log.Trace("Resolving dependency: %s", v)
+func (d *Dependency) Resolve(p *Module) error {
+	if gm, ok := global_modules[d.Name+"-"+d.Version]; ok {
+		log.Trace("Dependency %s already resolved (S1): %s", d, gm)
+		d.Module = gm
+		return nil
+	}
+
+	log.Trace("Resolving dependency: %s", d)
 
 	for _, s := range config.Sources {
 		log.Trace("=> Trying source: %s", s)
-		m, err := s.Find(v)
+		m, err := s.Find(d)
 		if err != nil {
 			log.Trace("=> Error from source: %s", err)
 			continue
 		}
 		if m != nil {
 			log.Trace("=> Resolved dependency: %s", m)
-			v.Module = m
-			return nil
+			d.Module = m
+			break
+		}
+	}
+	if d.Module == nil {
+		log.Error("Error resolving dependency: %s", d)
```

Comment:
d is not a string, and this message implies an error occurred when it didn't.

Perhaps `log.Error("Dependency [%s] not resolved", d.Name)`